src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorage.cs

```diff
@@ -0,0 +1,194 @@
+
+                throw new InvalidOperationException(SR.IsolatedStorage_ApplicationUndefined);
+            }
+        }
+
+        public object AssemblyIdentity
+        {
+            get
+            {
+                if (Helper.IsAssembly(Scope))
+                    return _assemblyIdentity;
+
+                throw new InvalidOperationException(SR.IsolatedStorage_AssemblyUndefined);
+            }
+        }
+
+        public object DomainIdentity
+        {
+            get
+            {
+                if (Helper.IsDomain(Scope))
+                    return _domainIdentity;
+
+                throw new InvalidOperationException(SR.IsolatedStorage_AssemblyUndefined);
+            }
+        }
+
+        [CLSCompliant(false)]
+        [Obsolete("IsolatedStorage.CurrentSize has been deprecated because it is not CLS Compliant.  To get the current size use IsolatedStorage.UsedSize")]
+        public virtual ulong CurrentSize
+        {
+            get
+            {
+                throw new InvalidOperationException(string.Format(SR.IsolatedStorage_CurrentSizeUndefined, nameof(CurrentSize)));
+            }
+        }
+
+        public virtual long UsedSize
+        {
+            get
+            {
+                throw new InvalidOperationException(string.Format(SR.IsolatedStorage_QuotaIsUndefined, nameof(UsedSize)));
+            }
+        }
+
+        public virtual long AvailableFreeSpace
+        {
+            get
+            {
+                throw new InvalidOperationException(string.Format(SR.IsolatedStorage_QuotaIsUndefined, nameof(AvailableFreeSpace)));
+            }
+        }
+
+        [CLSCompliant(false)]
+        [Obsolete("IsolatedStorage.MaximumSize has been deprecated because it is not CLS Compliant.  To get the maximum size use IsolatedStorage.Quota")]
+        public virtual ulong MaximumSize
+        {
+            get
+            {
+                if (_validQuota)
+                    return _quota;
+
+                throw new InvalidOperationException(string.Format(SR.IsolatedStorage_QuotaIsUndefined, nameof(MaximumSize)));
+            }
+        }
+
+        public virtual long Quota
+        {
+            get
+            {
+                if (_validQuota)
+                    return (long)_quota;
+
+                throw new InvalidOperationException(string.Format(SR.IsolatedStorage_QuotaIsUndefined, nameof(Quota)));
+            }
+
+            internal set
+            {
+                _quota = (ulong)value;
+                _validQuota = true;
+            }
+        }
+
+        public IsolatedStorageScope Scope
+        {
+            get; protected set;
+        }
+
+        protected virtual char SeparatorExternal
+        {
+            get { return Path.DirectorySeparatorChar; }
+        }
+
+        protected virtual char SeparatorInternal
+        {
+            get { return '.'; }
+        }
+
+        public virtual bool IncreaseQuotaTo(long newQuotaSize)
```

Comment:
Thanks for catching this- I'll override and return true. (`IsolatedStorage` returns `false` on desktop.)